libraries/vault_config.rb

```diff
@@ -32,21 +32,27 @@ class VaultConfig < Chef::Resource
 
       # @see https://vaultproject.io/docs/config/index.html
       attribute(:address, kind_of: String)
-      attribute(:tls_disable, kind_of: String, default: 'false')
+      attribute(:tls_disable, default: 'false', equal_to: %w{0 no false 1 yes true})
       attribute(:tls_cert_file, kind_of: String)
       attribute(:tls_key_file, kind_of: String)
       attribute(:bag_name, kind_of: String, default: 'secrets')
       attribute(:bag_item, kind_of: String, default: 'vault')
       attribute(:disable_mlock, equal_to: [true, false], default: false)
+      attribute(:default_lease_ttl, kind_of: String)
+      attribute(:max_lease_ttl, kind_of: String)
       attribute(:statsite_addr, kind_of: String)
       attribute(:statsd_addr, kind_of: String)
-      attribute(:backend_type, default: 'inmem', equal_to: %w{consul inmem zookeeper file})
+      attribute(:backend_type, default: 'inmem', equal_to: %w{consul etcd zookeeper dynamodb s3 mysql postgresql inmem file})
       attribute(:backend_options, option_collector: true)
-      attribute(:manage_certificate, kind_of: [TrueClass, FalseClass], default: true)
+      # Must be equal to one of: consul, etcd, zookeeper, dynamodb,
+      attribute(:habackend_type, kind_of: String)
+      attribute(:habackend_options, option_collector: true)
+      attribute(:manage_certificate, kind_of: [TrueClass, FalseClass], default: false)
 
       def tls?
-        return true unless %w{1 true}.include?(tls_disable) && manage_certificate
-
+        if manage_certificate
```

Comment:
Should we be evaluating this in this method? Couldn't the wrapper cookbook implement certificate management? I'm in favour of getting rid of this.